src/screens/SettingsScreen.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  StyleSheet,
+  Alert,
+  ActivityIndicator,
+  Image
+} from 'react-native';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { NavigationService } from '../utils/NavigationService';
+import { Storage } from '../utils/Storage';
+import { AccountManager } from '../account/AccountManager';
+import { RootStackParamList } from '../../App';
+import { headerStyles } from '../styles/HeaderStyles';
+
+type SettingsScreenProps = NativeStackScreenProps<RootStackParamList, 'Settings'>;
+
+export const SettingsScreen: React.FC<SettingsScreenProps> = ({ navigation }) => {
+  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    checkAuthStatus();
+  }, []);
+
+  const checkAuthStatus = async () => {
+    const authToken = await Storage.getAuthToken();
+    setIsSignedIn(Boolean(authToken));
+  };
+
+  const handleSignInPress = () => {
+    navigation.navigate('Login', { 
+      isFromTVSignIn: false,
+      onSignInComplete: () => {
+        checkAuthStatus();
+        navigation.goBack();
+      }
+    });
+  };
```

Comment:
Same here as well, navigation callback can be memorized

const handleSignInPress = useCallback(() => {
  navigation.navigate('Login', {
    isFromTVSignIn: false,
    onSignInComplete: () => {
      checkAuthStatus();
      navigation.goBack();
    }
  });
}, [navigation, checkAuthStatus]);

Reply:
DONE, I have updated it as Claude suggested, pls recheck.